ocs_ci/utility/rosa.py

```diff
@@ -0,0 +1,224 @@
+
+    Args:
+        version (str): cluster version
+        prefix (str): role prefix
+
+    """
+    version = get_semantic_version(version, only_major_minor=True)
+    cmd = (
+        f"rosa create account-roles --version {version} --mode auto"
+        f' --permissions-boundary "" --prefix {prefix}  --yes'
+    )
+    utils.run_cmd(cmd)
+
+
+def create_operator_roles(cluster, prefix='""'):
+    """
+    Create the cluster-specific Operator IAM roles. The roles created include the
+    relevant prefix for the cluster name
+
+    Args:
+        cluster (str): cluster name or cluster id
+        prefix (str): role prefix
+
+    """
+    cmd = (
+        f"rosa create operator-roles --cluster {cluster} --prefix {prefix}"
+        f' --mode auto --permissions-boundary "" --yes'
+    )
+    utils.run_cmd(cmd)
+
+
+def create_oidc_provider(cluster):
+    """
+    Create the OpenID Connect (OIDC) provider that the Operators will use to
+    authenticate
+
+    Args:
+        cluster (str): cluster name or cluster id
+
+    """
+    cmd = f"rosa create oidc-provider --cluster {cluster} --mode auto --yes"
+    utils.run_cmd(cmd)
+
+
+def download_rosa_cli():
+    """
+    Method to download OCM cli
+
+    Returns:
+        str: path to the installer
+
+    """
+    force_download = (
+        config.RUN["cli_params"].get("deploy")
+        and config.DEPLOYMENT["force_download_rosa_cli"]
+    )
+    return utils.get_rosa_cli(
+        config.DEPLOYMENT["rosa_cli_version"], force_download=force_download
+    )
+
+
+def get_addon_info(cluster, addon_name):
+    """
+    Get line related to addon from rosa `list addons` command.
+
+    Args:
+        cluster (str): cluster name
+        addon_name (str): addon name
+
+    Returns:
+        str: line of the command for relevant addon
+
+    """
+    cmd = f"rosa list addons -c {cluster}"
+    output = utils.run_cmd(cmd)
+    line = [line for line in output.splitlines() if re.match(f"^{addon_name} ", line)]
+    return line[0]
+
+
+def install_odf_addon(cluster):
+    """
+    Install ODF Managed Service addon to cluster.
+
+    Args:
+        cluster (str): cluster name or cluster id
+
+    """
+    addon_name = config.DEPLOYMENT["addon_name"]
+    size = config.ENV_DATA["size"]
+    notification_email_0 = config.ENV_DATA.get("notification_email_0")
+    notification_email_1 = config.ENV_DATA.get("notification_email_1")
+    notification_email_2 = config.ENV_DATA.get("notification_email_2")
+    cmd = f"rosa install addon --cluster={cluster} --size {size} {addon_name}" f" --yes"
+    if notification_email_0:
+        cmd = cmd + f" --notification-email-0 {notification_email_0}"
+    if notification_email_1:
+        cmd = cmd + f" --notification-email-1 {notification_email_1}"
+    if notification_email_2:
+        cmd = cmd + f" --notification-email-2 {notification_email_2}"
+
+    utils.run_cmd(cmd)
+    for addon_info in utils.TimeoutSampler(
+        10000, 30, get_addon_info, cluster, addon_name
```

Comment:
this is a very long time to wait. Is it intentional?

Reply:
I will update the time.